native/packages/react-native-bpk-styles/src/shadows.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PixelRatio } from 'react-native';
+import {
+  shadowSmColor,
+  shadowSmOffsetWidth,
+  shadowSmOffsetHeight,
+  shadowSmOpacity,
+  shadowSmRadius,
+  shadowXlColor,
+  shadowXlOffsetWidth,
+  shadowXlOffsetHeight,
+  shadowXlOpacity,
+  shadowXlRadius,
+} from 'bpk-tokens/tokens/base.react.native';
+
+const tokens = {
+  base: {
+    color: shadowSmColor,
+    offsetWidth: shadowSmOffsetWidth,
+    offsetHeight: shadowSmOffsetHeight,
+    opacity: shadowSmOpacity,
+    radius: shadowSmRadius,
+  },
+  large: {
+    color: shadowXlColor,
+    offsetWidth: shadowXlOffsetWidth,
+    offsetHeight: shadowXlOffsetHeight,
+    opacity: shadowXlOpacity,
+    radius: shadowXlRadius,
+  },
+};
+
+const getShadowStyles = type => ({
+  shadowColor: tokens[type].color,
+  shadowOffset: {
+    height: tokens[type].offsetHeight / PixelRatio.get(),
+    width: tokens[type].offsetWidth,
+  },
+  shadowOpacity: tokens[type].opacity,
+  shadowRadius: tokens[type].radius / PixelRatio.get(),
+});
+
+const shadows = {
+  base: () => getShadowStyles('base'),
+  large: () => getShadowStyles('large'),
+};
```

Comment:
Why not just:
```
export const base = () => ({
  shadowColor: shadowSmColor,
  shadowOffset: {
    height: shadowSmOffsetHeight / PixelRatio.get(),
    width: shadowSmOffsetWidth,
  },
  shadowOpacity: shadowSmOpacity,
  shadowRadius: shadowSmRadius / PixelRatio.get(),
});

export const large = () => ({
  shadowColor: shadowXlColor,
  shadowOffset: {
    height: shadowXlOffsetHeight / PixelRatio.get(),
    width: shadowXlOffsetWidth,
  },
  shadowOpacity: shadowXlOpacity,
  shadowRadius: shadowXlRadius / PixelRatio.get(),
});
```

Reply:
Just over-optimisation I guess 😅 . Updated now anyway, I'll merge it.